modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/authcode/UpdatingAuthorizationCodeState.java

```diff
@@ -55,14 +60,22 @@ private void update(ResourceOwnerOAuthContext context) {
         onUpdate.accept(context);
       }
     });
+    this.dancer = dancer;
```

Comment:
it's weird for this class to obtain the dancer. This is a state class, by doing this, you're turning it into a behavior one.

Reply:
replaced it with Supplier, instead of having dancer in it.